common/core/src/retry_operation.ts

```diff
@@ -37,6 +37,9 @@ export class RetryOperation {
       this._retryCount++;
       /*Codes_SRS_NODE_COMMON_RETRY_OPERATION_16_001: [The `operation` function should be called at every retry.]*/
       operation((err, result, response) => {
+        if (this._currentTimeout) {
+            this._currentTimeout = 0;
+        }
```

Comment:
Why is this here?

Reply:
I was getting a linter error saying that this._currentTimeout is set but never read and this was easier than figuring out the suppression syntax.  
As it turns out, it wasn't a linter error, it used to be a linter error, but now it's a compiler error.  I needed to set noUnusedLocals to false in tsconfig.json to suppress this error.

---
In reply to: [270512971](https://github.com/Azure/azure-iot-sdk-node/pull/521#discussion_r270512971) [](ancestors = 270512971)

Reply:
Why not simply do clearTimeout(this._currentTimeout);

---
In reply to: [270613536](https://github.com/Azure/azure-iot-sdk-node/pull/521#discussion_r270613536) [](ancestors = 270613536,270512971)